### PR TITLE
fix(bytetrack): fix uninteded roi value error due to casting int to uint

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

Sometimes bytetrack publish invalid roi with very large  x or y offset value like `4294967293`.
I found this is caused by mis-type casting from bytetrack roi `int32` to ros message roi `uint32`.

So I fixed message conversion part.

### <samp>🤖 Generated by Copilot at 7c8e94f</samp>

Fix bytetrack node crash when roi is outside of image. Adjust roi values to fit image boundaries and message type in `perception/bytetrack/src/bytetrack_node.cpp`.

## Tests performed

Tested with yolo roi and odaiba video image.
Original yolo roi do not publish invalid rois and after this PR, bytetrack also publish invalid rois.

Screenshot from 2023-11-15 15-31-14

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/bytetrack/src/bytetrack_node.cpp [hunk 1]
```diff
@@ -83,10 +83,18 @@ void ByteTrackNode::on_rect(
   bytetrack::ObjectArray objects = bytetrack_->update_tracker(object_array);
   for (const auto & tracked_object : objects) {
     tier4_perception_msgs::msg::DetectedObjectWithFeature object;
-    object.feature.roi.x_offset = tracked_object.x_offset;
-    object.feature.roi.y_offset = tracked_object.y_offset;
-    object.feature.roi.width = tracked_object.width;
-    object.feature.roi.height = tracked_object.height;
+    // fit xy offset to 0 if roi is outside of image
+    const int outside_x = 0 - tracked_object.x_offset;
+    const int outside_y = 0 - tracked_object.y_offset;
```
Comment (miursh) [inline perception/bytetrack/src/bytetrack_node.cpp:88]: [Q]
` std::max(-tracked_object.x_offset, 0)` is more understandable?

Comment (YoshiRi) [reply perception/bytetrack/src/bytetrack_node.cpp]: I agree. Fixed in the last commit.

**Review by miursh — COMMENTED** [review]
LGTM, but I have not confirmed it on the actual ECU.

**Review by miursh — APPROVED** [review]

**Comment by YoshiRi** [comment]
OK. This is just quick fix. So I will merge it.

**Review by manato — APPROVED** [review]
LGTM.
I confirmed that this modification can be built on a perception ECU. Thank you for the fixing @YoshiRi !

**Pull request merged**